src/main/java/DilemmaDetector/Simulator/SimulatorEngine.java

```diff
@@ -98,14 +101,27 @@ public SimulatorEngine(Model model, CollisionConsequencePredictor consequencePre
                         SIMULATION_TIME = updateSimulationTime(currentTime);
                     }
                 }
-                else if(!factoryWrapper.isPedestrian(actor)){
+                else if (factoryWrapper.isSurrounding(actor)){
+                    collisionWithSurrounding = true;
+                }
+                else if(factoryWrapper.isVehicle(actor)){
                     if (!actor.equals(mainVehicle)) {
-                        collisionNotWithPedestrian = true;
+                        collisionWithVehicle = true;
                     }
                 }
+                else if(factoryWrapper.isCollidableObstacle(actor.getEntityName())){
+                    collisionWithObstacle = true;
+                }
```

Comment:
Trochę bez sensu bo już mamy w Actorze zapisaną tą informację. Znaczy wiem, że to kategorię z tego chcesz wyciągnąć ale znowu sprawdzamy to samo. Może ok, może nie, w każdym razie I'm pointing out.

Reply:
Może trzeba by inną metodę do tego? Bo nonCollidableObstacle tu nie będziesz miał bo się nie zcollidują. isObstacle by w zupełności wystarczyło?

Reply:
Możliwe że tak, sprawdzę